include/smooth/feedback/qp_solver.hpp

```diff
@@ -583,15 +627,21 @@ class QPSolver
     // check primal
     Ax_.noalias()        = pbm.A * x_us_;
     const Scalar Ax_norm = norm(Ax_);
+    //static std::optional<Scalar> Ax_norm_old;
     Ax_ -= z_us_;
-    if (norm(Ax_) <= prm_.eps_abs + prm_.eps_rel * std::max<Scalar>(Ax_norm, norm(z_us_))) {
+    //std::cout << "primal check, norm(Ax_): " << norm(Ax_) << ", delta norm: " << (Ax_norm_old ? std::abs(Ax_norm - *Ax_norm_old) : 0.) << ",norm(z_us_): " << norm(z_us_) << ", eps: " << prm_.eps_abs + prm_.eps_rel * std::max<Scalar>(Ax_norm, norm(z_us_)) << std::endl;
+    //if (norm(Ax_) <= prm_.eps_abs + prm_.eps_rel * std::max<Scalar>(Ax_norm, norm(z_us_)) or (Ax_norm_old and std::abs(Ax_norm - *Ax_norm_old) < 1e-6)) {
+    if (norm(Ax_) <= prm_.eps_abs + prm_.eps_rel * std::max<Scalar>(Ax_norm, norm(z_us_)))
```

Comment:
The check here was not working, I had to look at Ax norm vs old Ax norm, I am not sure why.  I don't really care about that though, the idea is govern step size.  In any case this check will probably fail if you don't put a max time or max iter or check delta objective in a dumber way.